torchgeo/models/cnnlstm.py

```diff
@@ -0,0 +1,378 @@
+        ValueError: If the output dimension of CNN does not match input dimension of LSTM,
+                    or if the output dimension of LSTM does not match input dimension of task head.
+
+    Forward:
+        x (torch.Tensor): Input tensor, either
+            - (B, T, C, H, W) sequence of images, or
+            - (B, C, H, W) single image.
+        lengths (torch.Tensor, optional): Lengths of sequences for LSTM packing.
+
+    Returns:
+        Dict[str, Optional[torch.Tensor]]: Output dictionary from task head.
+    """
+
+    def __init__(self, cnn_backbone: CNN, rnn_encoder: LSTM, task_head: Task) -> None:
+        """Initialize CNN-LSTM model.
+
+        Args:
+            cnn_backbone: CNN feature extractor module.
+            rnn_encoder: LSTM sequence encoder module.
+            task_head: Task head for classification or regression.
+
+        Raises:
+            ValueError: If the output dimension of CNN does not match input dimension of LSTM,
+                        or if the output dimension of LSTM does not match input dimension of task head.
+        """
+        super().__init__()
+        self.cnn_backbone = cnn_backbone
+        self.rnn_encoder = rnn_encoder
+        self.task_head = task_head
+
+        # Validate CNN -> RNN dimensions
+        if cnn_backbone.output_dim != rnn_encoder.input_dim:
+            raise ValueError(
+                f'CNN output dim ({cnn_backbone.output_dim}) must match '
+                f'RNN input dim ({rnn_encoder.input_dim})'
+            )
+
+        rnn_output_dim = rnn_encoder.output_dim
+        task_input_dim = task_head.head.in_features
+
+        if rnn_output_dim != task_input_dim:
+            raise ValueError(
+                f'RNN output dim ({rnn_output_dim}) must match '
+                f'Task input dim ({task_input_dim})'
+            )
+
+    def forward(
+        self, x: torch.Tensor, lengths: torch.Tensor | None = None, **kwargs: Any
+    ) -> dict[str, torch.Tensor | None]:
+        """Forward pass through the complete CNN-LSTM model.
+
+        Args:
+            x: Input tensor, either (B, T, C, H, W) sequence of images or
+               (B, C, H, W) single image.
+            lengths: Lengths of sequences for LSTM packing (optional).
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            Output dictionary from task head containing predictions and probabilities.
+        """
+        # Handle single image input by adding time dimension
+        if x.dim() == 4:  # (B, C, H, W)
+            x = x.unsqueeze(1)  # (B, 1, C, H, W)
+
+        # CNN feature extraction (supports batching all frames)
+        cnn_features = self.cnn_backbone(x)  # (B, T, feature_dim)
+
+        # Sequence modeling with LSTM
+        rnn_output = self.rnn_encoder(cnn_features, lengths)  # (B, T, hidden_dim)
+
+        # Task-specific output
+
+        output = self.task_head(rnn_output)
+
+        return output  # type: ignore[no-any-return]
```

Comment:
Same here, use `x` as the variable name for all steps and you won't have to ignore the return type